src/executor/executor.pas

```diff
@@ -3009,7 +3009,8 @@ procedure TExecutor.ExecSelect(ST: TSelect);
   ST.ExecResult := csrFailed;
   try
     DoStatement(ST.Statement);
-    ST.ExecResult := csrSuccess;
+//  ExecResult for the select will be that of the statement executed
```

Comment:
This is not correct if we want the **select** statement to return the ExecResult of the executes sub-statements.

It should look something like:
ST.ExecResult := ST.Statement.ExecResult

But this must be checked if it hold in the case where there are multiple statements (TStatementList)

Most of all though. This change belongs in it's own Pull Request since the title is about graph command not failing correctly and this change has nothing to do with graph (directly).

Reply:
I'll revert this change and resubmit. I wasn't thinking about multiple statements. This is an added level of complexity. I'm thinking that if any statement in the do begin ... end block fails, the select should fail.

Reply:
I also think that should be the approach. I'm just not sure the code is made that way though.

Reply:
I'll put some more thought into it. I think it is primarily an issue for getting the correct result with an assert. We would never build a validation script with a do begin ... end block, would we?
I'll play around with scripts to see what unexpected results we get with the current approach.

Reply:
I completely agreed that we would never do that. But the ExecResult is also used internally to stop execution under some circumstances. So it is a good idead to propagate a failed statement anyways.